File: Medium/1846. Maximum Element After Decreasing and Rearranging/1846. Maximum Element After Decreasing and Rearranging.py
from typing import List

class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()  # Step 1

        prev = 0
        for i in range(len(arr)):
            arr[i] = min(prev + 1, arr[i])  # Step 2
            prev = arr[i]

        return max(arr)  # Step 3

File: Medium/1846. Maximum Element After Decreasing and Rearranging/1846. Maximum Element After Decreasing and Rearranging.rs
impl Solution {
    pub fn maximum_element_after_decrementing_and_rearranging(mut arr: Vec<i32>) -> i32 {
        arr.sort(); // Step 1

        let mut prev = 0;
        for i in 0..arr.len() {
            arr[i] = arr[i].min(prev + 1); // Step 2
            prev = arr[i];
        }

        arr.iter().max().unwrap_or(&0).clone() // Step 3
    }
}
